src/lang-promql/complete/hybrid.ts

```diff
@@ -331,12 +346,34 @@ export function analyzeCompletion(state: EditorState, node: Subtree): Context[]
         // Get the labelName.
         // By definition it's the firstChild: https://github.com/promlabs/lezer-promql/blob/0ef65e196a8db6a989ff3877d57fd0447d70e971/src/promql.grammar#L250
         let labelName = '';
-        if (node.parent.firstChild && node.parent.firstChild.type.id === LabelName) {
+        if (node.parent.firstChild?.type.id === LabelName) {
           labelName = state.sliceDoc(node.parent.firstChild.start, node.parent.firstChild.end);
         }
         // then find the metricName if it exists
         const metricName = getMetricNameInVectorSelector(node, state);
-        result.push({ kind: ContextKind.LabelValue, metricName: metricName, labelName: labelName });
+        const context = {
+          kind: ContextKind.LabelValue,
+          metricName: metricName,
+          labelName: labelName,
+          filters: [] as string[],
+        };
+        // In addition, it is interesting to know if there is other labelMatcher in the current VectorSelector.
```

Comment:
I wonder if it's worth adding this extra complexity, as the number of possible label names on a given metric is already small anyway, and at least in principle, it is legal to use the same label name multiple times, with different matching conditions. For example, it is not common, but you could do something like `foo{fruit!="banana",fruit!="apple"}` (of course you could also shorten that with a negative regex selector).

Reply:
Actually it's only for the labelValue not the labelName.
If I'm following your example, then when you would start to autocomplete the 2nd 'fruit' it won't return 'banana' because it's already used for the labelName fruit

Reply:
If you look at the gif in the PR description you will see that the label `instance` is still autocomplete but the value is not for the 2nd time

Reply:
Ah sorry, I didn't notice that it was doing it for the label value, not the name. In that case though it seems even more exotic because almost nobody repeats the same label name twice in a selector, so it adds a bit of code complexity for like a 1% use case.

Reply:
mmm yeah you are certainly right. I will remove the code then

Reply:
removed